### DISCUSSION TITLE
Can't create first test jail? FileNotFoundError: [Errno 2] No such file or directory: 'zfs'. SOLVED, sharing for others....

### DISCUSSION THREAD

**Discussion by b-neufeld** [Jip-Hop/jailmaker#156] [General]
Here's a dumb one. 

I upgraded to Dragonfish a couple days ago and thought I'd build by my first sandbox tonight. 

I followed the [TrueNAS Scale Sandbox instructions](https://www.truenas.com/docs/scale/scaletutorials/apps/sandboxes/) _to the letter_ to create a jailmaker dataset in the root of one of my pools (in fact I've tried both pools), then downloaded the jlmkr.py script, did the chmod command, then picked up the instructions on this Github project with the `./jlmkr.py install` command. All good so far. 

When I try to create the simple test jail...

``` 
root@ceres[/mnt/pallas/jailmaker]# jlmkr create --start myjail
USE THIS SCRIPT AT YOUR OWN RISK!
IT COMES WITHOUT WARRANTY AND IS NOT SUPPORTED BY IXSYSTEMS.
Creating jail myjail with default config.

TIP: Run `jlmkr create` without any arguments for interactive config.
Or use CLI args to override the default options.
For more info, run: `jlmkr create --help`

Creating ZFS Dataset pallas/jailmaker/jails
Traceback (most recent call last):
  File "/mnt/pallas/jailmaker/jlmkr.py", line 2150, in <module>
    main()
  File "/mnt/pallas/jailmaker/jlmkr.py", line 2145, in main
    sys.exit(func(**args))
             ^^^^^^^^^^^^
  File "/mnt/pallas/jailmaker/jlmkr.py", line 1477, in create_jail
    raise error
  File "/mnt/pallas/jailmaker/jlmkr.py", line 1321, in create_jail
    create_zfs_dataset(JAILS_DIR_PATH)
  File "/mnt/pallas/jailmaker/jlmkr.py", line 957, in create_zfs_dataset
    subprocess.run(["zfs", "create", dataset_to_create], check=True)
  File "/usr/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 1024, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.11/subprocess.py", line 1901, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'zfs'
root@ceres[/mnt/pallas/jailmaker]#
```

Banging my head against the wall. What am I doing different from the instructions and the video tutorial??!

Finally got it. 🤦‍♂️ I log into the shell via the Cmder (not the web GUI), and I habitually switch to root with just the `su` command. What I just learned thanks to Dr. Google is that `su -` is the command to switch to a TrueNAS CLI-friendly shell. 

Sigh. Moving on... Hope this helps some other beginner who had the bright idea to not use the web GUI!

**Comment by Jip-Hop** [comment]
Thanks for sharing. Yes I should update the README with the "how to become root" command. I always use `sudo su` and from then on use jailmaker.
